omg/entities/player.py

```diff
@@ -85,14 +111,30 @@ def update(self, mouse_x, mouse_y, delta_time):
         )
         self._regenerate_mana(delta_time)
 
-    def shoot(self):
-        """Shoots a projectile and informs the observes with an ProjectileShotEvent."""
-        if self.skills.current_size == 0 or self.current_mana < 20:
+        # Update combined elements
+        if len(self.to_be_combined_element_buffer) == 2:
+            # Get the name of the skill after combining elements
+            new_skill = "".join(self.to_be_combined_element_buffer[::])
+
+            # Update crafted skill slots
+            self._update_crafted_skill_slots(new_skill)
+
+            # Empty the element buffer
+            self.to_be_combined_element_buffer = []
+
+    def shoot(self, skill_name: str):
+        """Shoot a projectile and inform the observers."""
+        if skill_name in crafted_skill_dictionary.keys():
+            self.crafted_skill.set_skill_attributes(skill_name)
```

Comment:
This logic here depends on global variables which obscure how it works. `crafted_skill_dictionary` is imported both here and used in set_skill_attributes as a global variable. 

I would refactor `set_skill_attributes(skill_name)` to `set_skill_attributes(skill_dictionary)` such that it sets attributes based on the input, which makes it more clear and easily seen.

That means, the logic here can be changed to 
```suggestion
            skill_dict : Dict= crafted_skill_dictionary.get(skill_name, None)
            if skill_dict:
                self.crafted_skill.set_skill_attributes(skill_dict)
```